packages/dcos-integration-test/extra/test_metrics.py

```diff
@@ -134,6 +134,127 @@ def test_metrics_agents_statsd(dcos_api_session):
+    for up to 300 seconds.
+
+    """
+    response = dcos_api_session.session.request(
+        'GET', 'http://{}:61091/metrics'.format(node))
+    assert response.status_code == 200, 'Status code: {}'.format(response.status_code)
+    assert check_func(response) is True
+
+
+def test_task_metrics_metadata(dcos_api_session):
+    # install marathon on marathon
+    install_response = dcos_api_session.cosmos.install_package('marathon')
+    data = install_response.json()
+    dcos_api_session.marathon.wait_for_deployments_complete()
+
+    wait = True
+    mesos_id = node = ''
+    while wait:
+        master = dcos_api_session.masters[0]
+        state_response = dcos_api_session.get('/state', host=master, port=5050)
+        assert state_response.status_code == 200
+        state = state_response.json()
+        for framework in state['frameworks']:
+            if framework['name'] == 'marathon':
+                for task in framework['tasks']:
+                    if task['name'] == 'marathon-user':
+                        mesos_id = task['slave_id']
+                        wait = False
+                        break
+                break
+
+    for agent in state['slaves']:
+        if agent['id'] == mesos_id:
+            node = agent['hostname']
+            break
+
+    def check_task_metrics(response):
+        for line in response.text.splitlines():
+            if '#' in line:
+                continue
+            if 'marathon-user' in line:
+                return ('service_name="marathon"' in line and
+                        'task_name="marathon-user"' in line)
+        return False
+    check_metrics_prom(dcos_api_session, node, check_task_metrics)
+
+    # uninstall and cleanup framework
+    dcos_api_session.cosmos.uninstall_package('marathon', app_id=data['appId'])
+
+
+def test_executor_metrics_metadata(dcos_api_session):
+    # get count of all running containers to check against later for complete teardown
+    num_containers = 0
+    for agent_ip in dcos_api_session.slaves:
+        response = dcos_api_session.metrics.get('/containers', node=agent_ip)
+        assert response.status_code == 200
+        num_containers += len(response.json())
+
+    try:
+        # install hello-world framework
+        install_response = dcos_api_session.cosmos.install_package('hello-world', package_version='2.2.0-0.42.2')
+        data = install_response.json()
+
+        dcos_api_session.marathon.wait_for_deployments_complete()
+
+        wait = True
+        mesos_id = ''
+        while wait:
+            master = dcos_api_session.masters[0]
+            state_response = dcos_api_session.get('/state', host=master, port=5050)
+            assert state_response.status_code == 200
+            state = state_response.json()
+            for framework in state['frameworks']:
+                if framework['name'] == 'marathon':
+                    for task in framework['tasks']:
+                        if task['name'] == 'hello-world':
```

Comment:
Wouldn't this get us the task that's running the hello-world framework, rather than the hello-world framework's own tasks? The tasks launched by the framework may not be scheduled on the same node as where the framework is running. Is the intention to look for the Marathon task running the framework, rather than one of the framework's own tasks? If that's the case, why do we need to install the hello-world package, instead of e.g. running a single-task Marathon app that only sleeps?

Reply:
I think this gets the same node that the hello-world executor is running on (I believe it is indicated by the node that the Marathon task running the framework is on)

Reply:
@philipnrmn to fact-check me :)